src/routers/nomination/index.ts

```diff
@@ -25,4 +30,30 @@ router.post(
   },
 );
 
+router.patch(
+  '/second',
+  getValidations(LocalRoutes.SECOND_NOMINEE),
+  validate,
+  auth(),
+  async (req: AuthenticatedRequest, res: Response, next: NextFunction) => {
+    try {
+      const nomination = await NominationModel.findOne({
+        position: req.query.position as string,
+        candidate: req.query.candidate as string,
+      });
+      if (!nomination) throw new Error('Could not find nomination');
+
+      const myId = req.user?._id;
+      if (nomination.seconds.includes(myId))
+        throw new Error('You already seconded this nomination');
+
+      nomination.seconds.push(myId);
```

Comment:
Can we do `nominations.seconds.addToSet(myId);`? It doesn't always work in sys because of Typegoose complexities (I think bc it doesn't implicitly type arrays as Mongoose arrays) but it should work here. Then we don't even have to check if it already includes seconds!

Reply:
I was taking a look at the https://mongoosejs.com/docs/schematypes.html and they don't have a `set` per say, but they do have a `map`. What do you think about using that instead of an `array`? What do you want to store the values as since the keys will be object ids.

Reply:
I was actually just thinking to use this:
https://mongoosejs.com/docs/api.html#mongoosearray_MongooseArray-addToSet

MongoDB has the `$addToSet` update operator which is applied to an array and makes it act like a set

Reply:
We may need to type the array differently in the schema to get access to the `addToSet` method. I'm not sure.

Reply:
updated in 037718410e26d9f8d1c9c24793ed8d96326eec43
#addToSet 🔥